### PR TITLE
Adding SFTPFileManager

### PR DESCRIPTION
**Pull request by djfreels**
Added a FileManager class for sftp.

### CODE REVIEW

File: spark-pipeline-engine/src/main/scala/com/acxiom/pipeline/fs/SFTPFileManager.scala [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    val chan = session.openChannel("sftp").asInstanceOf[ChannelSftp]
+    chan.setBulkRequests(bulkRequests)
+    chan
+  }
+
+  /**
+    * Connect to the file system
+    */
+  override def connect(): Unit = {
+    channel.connect()
```
Comment (dafreels) [inline spark-pipeline-engine/src/main/scala/com/acxiom/pipeline/fs/SFTPFileManager.scala]: Add a check here to see if there is already a connection

File: spark-pipeline-engine/src/test/scala/com/acxiom/pipeline/fs/FileManagerTests.scala [hunk 2]
```diff
@@ -171,4 +176,112 @@ class FileManagerTests extends FunSpec with Suite {
       sparkSession.stop()
     }
   }
+
+  describe("FileManager - SFTP") {
```
Comment (dafreels) [inline spark-pipeline-engine/src/test/scala/com/acxiom/pipeline/fs/FileManagerTests.scala]: Add a copy test to ensure that data can be read/written using the streams.

File: spark-pipeline-engine/src/main/scala/com/acxiom/pipeline/fs/SFTPFileManager.scala (hunk 1 above)
Comment (djfreels) [reply spark-pipeline-engine/src/main/scala/com/acxiom/pipeline/fs/SFTPFileManager.scala]: Done

File: spark-pipeline-engine/src/test/scala/com/acxiom/pipeline/fs/FileManagerTests.scala (hunk 2 above)
Comment (djfreels) [reply spark-pipeline-engine/src/test/scala/com/acxiom/pipeline/fs/FileManagerTests.scala]: I've added the test.

**Pull request merged**